airflow-core/src/airflow/api_fastapi/execution_api/versions/v2025_05_21.py

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Optional
+
+from cadwyn import VersionChange, schema
+from pydantic import Field
+
+from airflow.api_fastapi.execution_api.routes.xcoms import GetXComSliceFilterParams
+
+
+class RemoveGetXcomFilterParamsOffset(VersionChange):
+    """Remove 'offset' from GetXcomFilterParams."""
+
+    description = __doc__
+
+    instructions_to_migrate_to_previous_version = (
+        schema(GetXComSliceFilterParams)
+        .field("offset")
+        .existed_as(type=Optional[int], info=Field(default=None)),
+    )
```

Comment:
We will also need to remove it here right? https://github.com/apache/airflow/pull/50117/files#diff-f7c9ab9ad4675e23743cffb626e7a9c4ce28ceffef5dd0b6b6a5268fe18909c5R133

Reply:
I _think_ the idea of cadwyn is you should maintain backward compatibility in code, but remove the interface from spec with the migration? I’m honestly not sure.

Reply:
I think the problem is that older sdk clients would use the old endpoint when they do (e.g. `seq[0]`), so we need to keep the `offset` param working for them.